app/services/reddit_json_from_url_service.rb

```diff
@@ -0,0 +1,58 @@
+class RedditJsonFromUrlService
+  include ActionView::Helpers::SanitizeHelper
+
+  def initialize(url)
+    @url = ActionController::Base.helpers.strip_tags(url).strip
+  end
+
+  def parse
+    validate_url
+
+    # Requests to Reddit require a custom `User-Agent` header to prevent 429 errors
+    json = HTTParty.get("#{@url}.json", headers: { "User-Agent" => "ThePracticalDev" })
+
+    # The JSON response is an array with two items.
+    # The first one is the post itself, the second one are the comments
+    data = json.first["data"]["children"][0]["data"]
+
+    {
+      author: data["author"],
+      title: data["title"],
+      post_url: @url,
+      created_at: Time.zone.at(data["created_utc"]).strftime("%b %e '%y"),
+      post_hint: data["post_hint"],
+      image_url: data["url"],
+      thumbnail: data["thumbnail"],
+      selftext: parse_markdown_content(data["selftext"]),
+      selftext_html: data["selftext_html"]
+    }
+  end
+
+  private
+
+  def parse_markdown_content(content)
+    markdown = Redcarpet::Markdown.new(Redcarpet::Render::HTMLRouge, REDCARPET_CONFIG)
+    text = markdown.render(content)
+
+    truncated = HTML_Truncator.truncate(text, 60)
+    sanitize(truncated)
+  end
+
+  def validate_url
+    cleaned_url = @url.delete(" ")
+
+    return true if valid_url?(cleaned_url) && link_exists?
+
+    raise StandardError, "Invalid Reddit link: #{@url}"
+  end
+
+  def valid_url?(url)
+    url = URI.parse(url)
```

Comment:
I think we should also employ a regexp for reddit URLs, as we do in other liquid tags, for example 

https://github.com/thepracticaldev/dev.to/blob/1a40e10df8d82ed06d4c37cb5a05f4ea71efed0d/app/liquid_tags/codepen_tag.rb#L3

maybe not as complex but enought to not just validate anything thrown at it. 

Otherwise you get the following:

![Screenshot_2020-05-28 New Post - DEV(local) Community 👩‍💻👨‍💻](https://user-images.githubusercontent.com/146201/83165017-3db2db80-a10d-11ea-8f31-84777c71c428.png)

Reply:
I did give it a try but a hard time figuring out a good regex 😅  Could you, please give me an idea for this?

A valid Reddit URL usually looks like this:

```
https://www.reddit.com/r/IAmA/comments/afvl2w
```

Reply:
Yeah, this is though, also because it could also be https://reddit.com/r/IAmA/comments/afvl2w or https://www.reddit.com/r/IAmA/comments/afvl2w/im_scott_from_scotts_cheap_flights_my_profession/ or I guess any combination of the subreddits?

What if we just check for `\Ahttps://(www.)?reddit.com` ? or something like that?

Reply:
Basically: starts with https://reddit.com or https://www.reddit.com